internal/recover/recover.go

```diff
@@ -235,23 +224,32 @@ func CreateDatabaseBackup(filesystem *sys.OS) error {
 
 	logger.Info("Creating database backup", logger.Ctx{"archive": backupFilePath})
 
-	dbFS := os.DirFS(filesystem.StateDir)
-	dbFiles, err := fs.Glob(dbFS, "database/*")
+	// For DB backups the tarball should contain `database/` so that the user
+	// can easily untar the backup from the state dir.
+	// Forget ^ if DatabaseDir isn't inside StateDir
+	rootDir := filesystem.StateDir
+	walkDir, err := filepath.Rel(filesystem.StateDir, filesystem.DatabaseDir)
```

Comment:
Not sure if more performant but why not using `filepath.Base(filesystem.DatabaseDir)`? It should give the same result and maybe is more intuitive?

Had to first check in the upstream docs what `Rel` is doing :) Not a blocker though.

Reply:
`filepath.Base` would assume that `DatabaseDir` is a directory inside `StateDir`. While I guess this is already sorta making that assumption, I wanted to capture any corner cases like:
```go
DatabaseDir = path.Join(StateDir, "somedir/database")
```

I clarified the comments a bit.

Reply:
I see, thanks.